File: 2019-12-5/TestTransition/TestTransition/ViewController.m
//
//  ViewController.m
//  TestTransition
//
//  Created by 吉腾蛟 on 2019/12/17.
//  Copyright © 2019 JY. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    NSNumber *number=@(123124);
    NSNumber *emptynumber;
    
    if ([number boolValue]) {
        NSLog(@"number 不为空");
    }
    if (![emptynumber boolValue]) {
        NSLog(@"emptynumber 为空");
    }
}


@end

File: 2019-5-29(SWift)/Persistence/Persistence/ViewController.swift
//
//  ViewController.swift
//  Persistence
//
//  Created by 吉腾蛟 on 2019/9/19.
//  Copyright © 2019 JY. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet var lineFields: [UITextField]!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        let filePath = self.dataFilePath()
        if FileManager.default.fileExists(atPath: filePath) {
            let array = NSArray(contentsOfFile: filePath) as! [String]
            for index in 0..<array.count {
                lineFields[index].text = array[index]
            }
        }
        let app = UIApplication.shared
        NotificationCenter.default.addObserver(self, selector: #selector(applicationWillResignActive(notification:)), name: NSNotification.Name(rawValue: "UIApplicationWillResignActiveNotification"), object: app)
        
    }

    func dataFilePath() -> String {
        let paths = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)
        let documentDirectory = paths[0] as NSString
        return documentDirectory.appendingPathComponent("data.plist") as String
        
    }
    
    
    @objc func applicationWillResignActive(notification: NSNotification) {
        let filePath = self.dataFilePath()
        let array = (self.lineFields as NSArray).value(forKey: "text") as! NSArray
        array.write(toFile: filePath, atomically: true)
    }
}

